packages/admin-ui/ui/src/components/templates/reservations-table/index.tsx

```diff
@@ -0,0 +1,358 @@
+
+  const updateUrlFromFilter = (obj = {}) => {
+    const stringified = qs.stringify(obj)
+    window.history.replaceState(`/a/reservations`, "", `${`?${stringified}`}`)
+  }
+
+  const refreshWithFilters = () => {
+    const filterObj = representationObject
+
+    if (isEmpty(filterObj)) {
+      updateUrlFromFilter({ offset: 0, limit: DEFAULT_PAGE_SIZE })
+    } else {
+      updateUrlFromFilter(filterObj)
+    }
+  }
+
+  useEffect(() => {
+    refreshWithFilters()
+  }, [representationObject])
+
+  const [columns] = useReservationsTableColumns()
+
+  const {
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    rows,
+    prepareRow,
+    gotoPage,
+    canPreviousPage,
+    canNextPage,
+    pageCount,
+    nextPage,
+    previousPage,
+    // Get the state from the instance
+    state: { pageIndex },
+  } = useTable(
+    {
+      columns,
+      data: reservations || [],
+      manualPagination: true,
+      initialState: {
+        pageIndex: Math.floor(offs / limit),
+        pageSize: limit,
+      },
+      pageCount: numPages,
+      autoResetPage: false,
+    },
+    usePagination
+  )
+
+  // Debounced search
+  useEffect(() => {
+    const delayDebounceFn = setTimeout(() => {
+      if (query) {
+        setFreeText(query)
+        gotoPage(0)
+      } else {
+        if (typeof query !== "undefined") {
+          // if we delete query string, we reset the table view
+          reset()
+        }
+      }
+    }, 400)
+
+    return () => clearTimeout(delayDebounceFn)
+  }, [query])
+
+  const handleNext = () => {
+    if (canNextPage) {
+      paginate(1)
+      nextPage()
+    }
+  }
+
+  const handlePrev = () => {
+    if (canPreviousPage) {
+      paginate(-1)
+      previousPage()
+    }
+  }
+
+  return (
+    <>
+      <TableContainer
+        hasPagination
+        pagingState={{
+          count: count || 0,
+          offset: offs,
+          pageSize: offs + rows.length,
+          title: "Reservations",
+          currentPage: pageIndex + 1,
+          pageCount: pageCount,
+          nextPage: handleNext,
+          prevPage: handlePrev,
+          hasNext: canNextPage,
+          hasPrev: canPreviousPage,
+        }}
+        numberOfRows={limit}
+        isLoading={isLoading}
+      >
+        <Table
+          enableSearch
+          searchClassName="h-[40px]"
+          handleSearch={setQuery}
+          searchValue={query}
+          tableActions={
+            <div className="flex gap-2">
+              <LocationDropdown
+                selectedLocation={
+                  queryObject.location_id || store?.default_location_id
+                }
+                onChange={(id) => {
+                  setLocationFilter(id)
+                  gotoPage(0)
+                }}
+              />
+              <Button
+                variant="secondary"
+                size="small"
+                onClick={openReservationCreate}
+              >
+                Create reservation
```

Comment:
thought(non-blocking): Would it be possible to default the location id in the "Create reservation"-view to the one selected in the table? It would make for a better admin experience.

Reply:
done! 😄